packages/flutter/lib/src/widgets/modal_button_sheet.dart

```diff
@@ -0,0 +1,652 @@
+  switch (platform) {
+    case TargetPlatform.iOS:
+    case TargetPlatform.android:
+      return PointerDeviceKind.touch;
+    case TargetPlatform.linux:
+    case TargetPlatform.macOS:
+    case TargetPlatform.windows:
+      return PointerDeviceKind.mouse;
+    case TargetPlatform.fuchsia:
+      return PointerDeviceKind.unknown;
+  }
+  return PointerDeviceKind.unknown;
+}
+
+// Copied from bottom_sheet.dart as is a private class
+// https://github.com/flutter/flutter/issues/51627
+
+// TODO(guidezpl): Look into making this public. A copy of this class is in
+//  scaffold.dart, for now, https://github.com/flutter/flutter/issues/51627
+
+/// A curve that progresses linearly until a specified [startingPoint], at which
+/// point [curve] will begin. Unlike [Interval], [curve] will not start at zero,
+/// but will use [startingPoint] as the Y position.
+///
+/// For example, if [startingPoint] is set to `0.5`, and [curve] is set to
+/// [Curves.easeOut], then the bottom-left quarter of the curve will be a
+/// straight line, and the top-right quarter will contain the entire contents of
+/// [Curves.easeOut].
+///
+/// This is useful in situations where a widget must track the user's finger
+/// (which requires a linear animation), and afterwards can be flung using a
+/// curve specified with the [curve] argument, after the finger is released. In
+/// such a case, the value of [startingPoint] would be the progress of the
+/// animation at the time when the finger was released.
+///
+/// The [startingPoint] and [curve] arguments must not be null.
+class BottomSheetSuspendedCurve extends ParametricCurve<double> {
+  /// Creates a suspended curve.
+  const BottomSheetSuspendedCurve(
+    this.startingPoint, {
+    this.curve = Curves.easeOutCubic,
+  })  : assert(startingPoint != null),
+        assert(curve != null);
+
+  /// The progress value at which [curve] should begin.
+  ///
+  /// This defaults to [Curves.easeOutCubic].
+  final double startingPoint;
+
+  /// The curve to use when [startingPoint] is reached.
+  final Curve curve;
+
+  @override
+  double transform(double t) {
+    assert(t >= 0.0 && t <= 1.0);
+    assert(startingPoint >= 0.0 && startingPoint <= 1.0);
+
+    if (t < startingPoint) {
+      return t;
+    }
+
+    if (t == 1.0) {
+      return t;
+    }
+
+    final curveProgress = (t - startingPoint) / (1 - startingPoint);
+    final transformed = curve.transform(curveProgress);
+    return lerpDouble(startingPoint, 1, transformed) ?? 0;
+  }
+
+  @override
+  String toString() {
+    return '${describeIdentity(this)}($startingPoint, $curve)';
+  }
+}
+
+/// Associates a [ScrollController] with a subtree.
+///
+/// This mechanism can be used to provide default behavior for scroll views in a
+/// subtree inside a modal bottom sheet.
+///
+/// We want to remove this and use [PrimaryScrollController].
+/// This issue should be solved first https://github.com/flutter/flutter/issues/64236
```

Comment:
I think this issue should be solved so Scaffold can inherit the `PrimaryScrollController` and `ModalScrollController` should not be needed.